Build failure when attempting to build from source

Unfortunately I wasn't able to build `staticx` from source for Python 3.6 inside a virtual environment (created via `pipenv`). It appears that `scons` might be misbehaving. Below is the output of `pip install .`:

```
$ pip install .
Processing /home/chris/start/stack/staticx
Requirement already satisfied: pyelftools in /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages (from staticx==0.5.0+13.g9545976.dirty) (0.25)
Building wheels for collected packages: staticx
  Running setup.py bdist_wheel for staticx ... error
  Complete output from command /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-n2ft99ij/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-vaoo_489 --python-tag cp36:
  /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/dist.py:398: UserWarning: Normalizing '0.5.0+13-g9545976-dirty' to '0.5.0+13.g9545976.dirty'
    normalized_version,
  running bdist_wheel
  running build
  running build_bootloader
  scons: Reading SConscript files ...
  scons: done reading SConscript files.
  scons: Building targets ...
  gcc -o scons_build/bootloader/error.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Ilibtar -Ilibxz bootloader/error.c
  gcc -o scons_build/bootloader/elfutil.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Ilibtar -Ilibxz bootloader/elfutil.c
  gcc -o scons_build/bootloader/extract.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Ilibtar -Ilibxz bootloader/extract.c
  gcc -o scons_build/bootloader/main.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Ilibtar -Ilibxz bootloader/main.c
  gcc -o scons_build/bootloader/mmap.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Ilibtar -Ilibxz bootloader/mmap.c
  gcc -o scons_build/bootloader/util.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -static -Ilibtar -Ilibxz bootloader/util.c
  gcc -o scons_build/libtar/append.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/append.c
  gcc -o scons_build/libtar/block.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/block.c
  gcc -o scons_build/libtar/decode.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/decode.c
  gcc -o scons_build/libtar/encode.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/encode.c
  libtar/encode.c: In function 'th_set_device':
  libtar/encode.c:148:13: error: In the GNU C Library, "major" is defined
   by <sys/sysmacros.h>. For historical compatibility, it is
   currently defined by <sys/types.h> as well, but we plan to
   remove this soon. To use "major", include <sys/sysmacros.h>
   directly. If you did not intend to use a system-defined macro
   "major", you should undefine it after including <sys/types.h>. [-Werror]
    int_to_oct(major(device), t->th_buf.devmajor, 8);
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  libtar/encode.c:149:13: error: In the GNU C Library, "minor" is defined
   by <sys/sysmacros.h>. For historical compatibility, it is
   currently defined by <sys/types.h> as well, but we plan to
   remove this soon. To use "minor", include <sys/sysmacros.h>
   directly. If you did not intend to use a system-defined macro
   "minor", you should undefine it after including <sys/types.h>. [-Werror]
    int_to_oct(minor(device), t->th_buf.devminor, 8);
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  cc1: all warnings being treated as errors
  scons: *** [scons_build/libtar/encode.o] Error 1
  scons: building terminated because of errors.
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-req-build-n2ft99ij/setup.py", line 95, in <module>
      'build':            build_hook,
    File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/__init__.py", line 140, in setup
      return distutils.core.setup(**attrs)
    File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/wheel/bdist_wheel.py", line 202, in run
      self.run_command('build')
    File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-req-build-n2ft99ij/setup.py", line 28, in run
      self.run_command('build_bootloader')
    File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-req-build-n2ft99ij/setup.py", line 23, in run
      check_call(['scons'])
    File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['scons']' returned non-zero exit status 2.
  
  ----------------------------------------
  Failed building wheel for staticx
  Running setup.py clean for staticx
Failed to build staticx
Installing collected packages: staticx
  Found existing installation: staticx 0.5.0
    Uninstalling staticx-0.5.0:
      Successfully uninstalled staticx-0.5.0
  Running setup.py install for staticx ... error
    Complete output from command /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-n2ft99ij/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-ncx02vel/install-record.txt --single-version-externally-managed --compile --install-headers /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/include/site/python3.6/staticx:
    /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/dist.py:398: UserWarning: Normalizing '0.5.0+13-g9545976-dirty' to '0.5.0+13.g9545976.dirty'
      normalized_version,
    running install
    running build
    running build_bootloader
    scons: Reading SConscript files ...
    scons: done reading SConscript files.
    scons: Building targets ...
    gcc -o scons_build/libtar/encode.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/encode.c
    libtar/encode.c: In function 'th_set_device':
    libtar/encode.c:148:13: error: In the GNU C Library, "major" is defined
     by <sys/sysmacros.h>. For historical compatibility, it is
     currently defined by <sys/types.h> as well, but we plan to
     remove this soon. To use "major", include <sys/sysmacros.h>
     directly. If you did not intend to use a system-defined macro
     "major", you should undefine it after including <sys/types.h>. [-Werror]
      int_to_oct(major(device), t->th_buf.devmajor, 8);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    libtar/encode.c:149:13: error: In the GNU C Library, "minor" is defined
     by <sys/sysmacros.h>. For historical compatibility, it is
     currently defined by <sys/types.h> as well, but we plan to
     remove this soon. To use "minor", include <sys/sysmacros.h>
     directly. If you did not intend to use a system-defined macro
     "minor", you should undefine it after including <sys/types.h>. [-Werror]
      int_to_oct(minor(device), t->th_buf.devminor, 8);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    cc1: all warnings being treated as errors
    scons: *** [scons_build/libtar/encode.o] Error 1
    scons: building terminated because of errors.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-n2ft99ij/setup.py", line 95, in <module>
        'build':            build_hook,
      File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/__init__.py", line 140, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib/python3.6/distutils/command/install.py", line 589, in run
        self.run_command('build')
      File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/tmp/pip-req-build-n2ft99ij/setup.py", line 28, in run
        self.run_command('build_bootloader')
      File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/tmp/pip-req-build-n2ft99ij/setup.py", line 23, in run
        check_call(['scons'])
      File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['scons']' returned non-zero exit status 2.
    
    ----------------------------------------
  Rolling back uninstall of staticx
Command "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-n2ft99ij/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-ncx02vel/install-record.txt --single-version-externally-managed --compile --install-headers /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/include/site/python3.6/staticx" failed with error code 1 in /tmp/pip-req-build-n2ft99ij/
```

Thanks for reporting. This appears to be an actual bug, albeit with `libtar` and not `staticx` itself.

This warning must have been added to a newer version of GLIBC than I have. What version of GLIBC are you using? Or what distro of Linux are you running?

```
    libtar/encode.c: In function 'th_set_device':
    libtar/encode.c:148:13: error: In the GNU C Library, "major" is defined
     by <sys/sysmacros.h>. For historical compatibility, it is
     currently defined by <sys/types.h> as well, but we plan to
     remove this soon. To use "major", include <sys/sysmacros.h>
     directly. If you did not intend to use a system-defined macro
     "major", you should undefine it after including <sys/types.h>. [-Werror]
      int_to_oct(major(device), t->th_buf.devmajor, 8);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

As the warning indicates, the problem is that libtar is using the `major()` and `minor()` macros [here](https://github.com/JonathonReinhart/staticx/blob/9545976bc53d28017171bd0190fd30caf91a9581/libtar/encode.c#L148), but only including `<sys/types.h>` and not `<sys/sysmacros.h>`.

[`MAKEDEV(3)`](http://man7.org/linux/man-pages/man3/makedev.3.html) shows including `<sys/sysmacros.h>` so I think the fix is to simply add that `#include` line:


```diff
diff --git a/libtar/encode.c b/libtar/encode.c
index 7671e4b..49ddb7f 100644
--- a/libtar/encode.c
+++ b/libtar/encode.c
@@ -14,6 +14,7 @@
 #include <pwd.h>
 #include <grp.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <string.h>
 #include <stdlib.h>
 #include "libtar.h"
```

Could you try that out and let me know if you can build staticx with your native glibc?

```
$ ldd --version
ldd (Ubuntu GLIBC 2.26-0ubuntu2.1) 2.26
...
```

Applying the patch didn't seem to fix the problem:
```
Processing /home/chris/start/stack/staticx
Requirement already satisfied: pyelftools in /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages (from staticx==0.5.0+13.g9545976.dirty) (0.25)
Building wheels for collected packages: staticx
  Running setup.py bdist_wheel for staticx ... error
  Complete output from command /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-zvan2nuv/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-xleuz6kt --python-tag cp36:
  /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/dist.py:398: UserWarning: Normalizing '0.5.0+13-g9545976-dirty' to '0.5.0+13.g9545976.dirty'
    normalized_version,
  running bdist_wheel
  running build
  running build_bootloader
  scons: Reading SConscript files ...
  scons: done reading SConscript files.
  scons: Building targets ...
  gcc -o scons_build/libtar/encode.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/encode.c
  gcc -o scons_build/libtar/extract.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/extract.c
  libtar/extract.c: In function 'tar_extract_chardev':
  libtar/extract.c:401:13: error: In the GNU C Library, "makedev" is defined
   by <sys/sysmacros.h>. For historical compatibility, it is
   currently defined by <sys/types.h> as well, but we plan to
   remove this soon. To use "makedev", include <sys/sysmacros.h>
   directly. If you did not intend to use a system-defined macro
   "makedev", you should undefine it after including <sys/types.h>. [-Werror]
       makedev(devmaj, devmin)) == -1)
               ^~~~~~~~~~~~~~~~~~~~~~~
  libtar/extract.c: In function 'tar_extract_blockdev':
  libtar/extract.c:440:13: error: In the GNU C Library, "makedev" is defined
   by <sys/sysmacros.h>. For historical compatibility, it is
   currently defined by <sys/types.h> as well, but we plan to
   remove this soon. To use "makedev", include <sys/sysmacros.h>
   directly. If you did not intend to use a system-defined macro
   "makedev", you should undefine it after including <sys/types.h>. [-Werror]
       makedev(devmaj, devmin)) == -1)
               ^~~~~~~~~~~~~~~~~~~~~~~
  cc1: all warnings being treated as errors
  scons: *** [scons_build/libtar/extract.o] Error 1
  scons: building terminated because of errors.
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-req-build-zvan2nuv/setup.py", line 95, in <module>
      'build':            build_hook,
    File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/__init__.py", line 140, in setup
      return distutils.core.setup(**attrs)
    File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/wheel/bdist_wheel.py", line 202, in run
      self.run_command('build')
    File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-req-build-zvan2nuv/setup.py", line 28, in run
      self.run_command('build_bootloader')
    File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
      cmd_obj.run()
    File "/tmp/pip-req-build-zvan2nuv/setup.py", line 23, in run
      check_call(['scons'])
    File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['scons']' returned non-zero exit status 2.
  
  ----------------------------------------
  Failed building wheel for staticx
  Running setup.py clean for staticx
Failed to build staticx
Installing collected packages: staticx
  Found existing installation: staticx 0.5.0+13.g9545976.dirty
    Uninstalling staticx-0.5.0+13.g9545976.dirty:
      Successfully uninstalled staticx-0.5.0+13.g9545976.dirty
  Running setup.py install for staticx ... error
    Complete output from command /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-zvan2nuv/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-04fe6enj/install-record.txt --single-version-externally-managed --compile --install-headers /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/include/site/python3.6/staticx:
    /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/dist.py:398: UserWarning: Normalizing '0.5.0+13-g9545976-dirty' to '0.5.0+13.g9545976.dirty'
      normalized_version,
    running install
    running build
    running build_bootloader
    scons: Reading SConscript files ...
    scons: done reading SConscript files.
    scons: Building targets ...
    gcc -o scons_build/libtar/extract.o -c -std=gnu99 -Wall -Werror -Wmissing-prototypes -Wstrict-prototypes -DPACKAGE_VERSION=\"1.2.20\" -Ilibtar -Ilibxz -Iscons_build/libtar/compat -Ilibtar/compat libtar/extract.c
    libtar/extract.c: In function 'tar_extract_chardev':
    libtar/extract.c:401:13: error: In the GNU C Library, "makedev" is defined
     by <sys/sysmacros.h>. For historical compatibility, it is
     currently defined by <sys/types.h> as well, but we plan to
     remove this soon. To use "makedev", include <sys/sysmacros.h>
     directly. If you did not intend to use a system-defined macro
     "makedev", you should undefine it after including <sys/types.h>. [-Werror]
         makedev(devmaj, devmin)) == -1)
                 ^~~~~~~~~~~~~~~~~~~~~~~
    libtar/extract.c: In function 'tar_extract_blockdev':
    libtar/extract.c:440:13: error: In the GNU C Library, "makedev" is defined
     by <sys/sysmacros.h>. For historical compatibility, it is
     currently defined by <sys/types.h> as well, but we plan to
     remove this soon. To use "makedev", include <sys/sysmacros.h>
     directly. If you did not intend to use a system-defined macro
     "makedev", you should undefine it after including <sys/types.h>. [-Werror]
         makedev(devmaj, devmin)) == -1)
                 ^~~~~~~~~~~~~~~~~~~~~~~
    cc1: all warnings being treated as errors
    scons: *** [scons_build/libtar/extract.o] Error 1
    scons: building terminated because of errors.
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-zvan2nuv/setup.py", line 95, in <module>
        'build':            build_hook,
      File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/__init__.py", line 140, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/lib/python3.6/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib/python3.6/distutils/command/install.py", line 589, in run
        self.run_command('build')
      File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/tmp/pip-req-build-zvan2nuv/setup.py", line 28, in run
        self.run_command('build_bootloader')
      File "/usr/lib/python3.6/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/tmp/pip-req-build-zvan2nuv/setup.py", line 23, in run
        check_call(['scons'])
      File "/usr/lib/python3.6/subprocess.py", line 291, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['scons']' returned non-zero exit status 2.
    
    ----------------------------------------
  Rolling back uninstall of staticx
Command "/home/chris/.local/share/virtualenvs/py36-SOcOLDvI/bin/python3.6m -u -c "import setuptools, tokenize;__file__='/tmp/pip-req-build-zvan2nuv/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-04fe6enj/install-record.txt --single-version-externally-managed --compile --install-headers /home/chris/.local/share/virtualenvs/py36-SOcOLDvI/include/site/python3.6/staticx" failed with error code 1 in /tmp/pip-req-build-zvan2nuv/
```

--- a/libtar/encode.c
+++ b/libtar/encode.c
@@ -14,6 +14,7 @@
 #include <pwd.h>
 #include <grp.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <string.h>
 #include <stdlib.h>
 #include "libtar.h"

--- a/libtar/extract.c
+++ b/libtar/extract.c
@@ -14,6 +14,7 @@
 #include <string.h>
 #include <sys/param.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <fcntl.h>
 #include <errno.h>
 #include <utime.h>

